sql/src/main/java/org/apache/druid/sql/calcite/planner/DruidTypeSystem.java

```diff
@@ -120,7 +124,13 @@ public RelDataType deriveAvgAggType(
       final RelDataType argumentType
   )
   {
-    return RelDataTypeSystem.DEFAULT.deriveAvgAggType(typeFactory, argumentType);
+    // Widen all averages to 64-bits regardless of the size of the inputs.
+
+    if (SqlTypeName.INT_TYPES.contains(argumentType.getSqlTypeName())) {
+      return Calcites.createSqlTypeWithNullability(typeFactory, SqlTypeName.BIGINT, argumentType.isNullable());
+    } else {
+      return Calcites.createSqlTypeWithNullability(typeFactory, SqlTypeName.DOUBLE, argumentType.isNullable());
+    }
```

Comment:
I don't know what are the benefits of widening the return type of `AVG` aggregates - I wonder if this change is strictly necessary for the upgrade - or it could be done separately?